core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionUnitSuite.scala

```diff
@@ -54,6 +57,106 @@ class BlockManagerDecommissionUnitSuite extends SparkFunSuite with Matchers {
     }
   }
 
+  /**
+   * Validate a given configuration with the mocks.
+   * The fail variable controls if we expect migration to fail, in which case we expect
+   * a constant Long.MaxValue timestamp.
+   */
+  private def validateWithMocks(conf: SparkConf, bm: BlockManager,
+      migratableShuffleBlockResolver: MigratableResolver, fail: Boolean = false) = {
+    // Verify the decom manager handles this correctly
+    val bmDecomManager = new BlockManagerDecommissioner(conf, bm)
+    var previousTime = Long.MaxValue
+    try {
+      bmDecomManager.start()
+      eventually(timeout(10.second), interval(10.milliseconds)) {
+        val (currentTime, done) = bmDecomManager.lastMigrationInfo()
+        assert(done)
+        // Make sure the time stamp starts moving forward.
+        if (!fail && previousTime > currentTime) {
+          previousTime = currentTime
+          assert(false)
+        } else if (fail) {
+          assert(currentTime === Long.MaxValue)
+        }
+      }
+      if (!fail) {
+        // Wait 5 seconds and assert times keep moving forward.
+        Thread.sleep(5000)
+        val (currentTime, done) = bmDecomManager.lastMigrationInfo()
+        assert(done && currentTime > previousTime)
+      }
+    } finally {
+      bmDecomManager.stop()
```

Comment:
Just making sure I understand: Is this test also intending to test that the BM decommissioning thread cleanly exits on its own ? I forget whether that was easy to check or not.

Reply:
Okay, Although I don't see any changes to the K8s suite in this PR. I do see the wait for the executor removal in the BM integration suite in this PR.

Reply:
Your right, it's just covered in the BM integration suite. I was looking at changing the K8s suite as well but doing that could have made one of the parts not necessarily tested the way I wanted so I backed out those changes.

Reply:
Sounds good !